packages/core/src/utils/JsonTransformer.ts

```diff
@@ -1,23 +1,61 @@
+import type { Validate } from 'class-validator'
+
 import { instanceToPlain, plainToInstance } from 'class-transformer'
 
+import { ClassValidationError } from '../error/ClassValidationError'
+import { isValidationErrorArray } from '../error/ValidationErrorUtils'
+
+import { MessageValidator } from './MessageValidator'
+
+interface Validate {
+  validate?: boolean
+}
+
 export class JsonTransformer {
   public static toJSON<T>(classInstance: T) {
     return instanceToPlain(classInstance, {
       exposeDefaultValues: true,
     })
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  public static fromJSON<T>(json: any, Class: { new (...args: any[]): T }): T {
-    return plainToInstance(Class, json, { exposeDefaultValues: true })
+  public static async fromJSON<T>(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    json: any,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    Class: { new (...args: any[]): T },
+    { validate = true }: Validate
+  ): Promise<T> {
+    if (!validate) {
+      return plainToInstance(Class, json, { exposeDefaultValues: true })
+    } else {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const plainInstance = plainToInstance(Class, json, { exposeDefaultValues: true }) as any
+      try {
+        await MessageValidator.validate(plainInstance)
```

Comment:
I dont think this is async, I might be wrong though.

Reply:
`MessageValidator.validate` indeed isn't async, so we can probably keep all `JsonTransformer` methods sync :)

Reply:
It returns a promise so it is async... but maybe we can update it to use `validateSync` instead: https://github.com/typestack/class-validator#synchronous-validation

I agree it would be nice to keep this method sync.

Only downside is that it will ignore all async validators, but we don't have any async validators so we're probably good

Reply:
ALso agree that keeping this sync (non-async) is preferable. also for people already using afj so they don't have to change their code

Reply:
I'll have a go at using the sync validation from the suggested link